src/app/qgisapp.cpp

```diff
@@ -660,6 +661,18 @@ void QgisApp::onActiveLayerChanged( QgsMapLayer *layer )
   emit activeLayerChanged( layer );
 }
 
+void QgisApp::vectorLayerStyleLoaded( const QgsMapLayer::StyleCategories &categories )
+{
+  if ( categories.testFlag( QgsMapLayer::StyleCategory::Forms ) )
+  {
+    QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( sender() );
+    if ( vl && vl->isValid( ) )
```

Comment:
Question -- why does this require a valid layer? You could potentially load a style for a layer with a broken path, and still get the dependencies resolved...

Reply:
If the layer is invalid I see no point in resolving dependencies, because you will not able to use them anyways. But, if you fix the broken layer later, then the style is applied again and the resolution will be attempted.